src/panoptes/pocs/focuser/birger.py

```diff
@@ -249,13 +249,15 @@ def _send_command(self, command, response_length=None):
         # Clear the input buffer in case there's anything left over in there.
         self._serial_port.reset_input_buffer()
 
-        # Send command
-        self._serial_io.write(command + '\r')
-
         # In verbose mode adaptor will first echo the command
-        echo = self._serial_io.readline().rstrip()
-        assert echo == command, self.logger.warning("echo != command: {} != {}".format(
-            echo, command))
+        echo = [self._serial_io.readline().rstrip()]
+        while echo[0] != command:
+            # Send command
+            self._serial_io.write(command + '\r')
+            echo[0] = self._serial_io.readline().rstrip()
```

Comment:
If the Birger has a problem worse than intermittent serial comms error then this will be an infinite loop! You should _always_ have a finite number of retries, with a small default value, and a corresponding config parameter to change it.

If the expected response is not received within the set number of retries then the code should break out of the loop, log an error, and throw an exception.